client/landing/stepper/declarative-flow/internals/hooks/use-sign-up-start-tracking/index.tsx

```diff
@@ -11,16 +11,50 @@ interface Props {
 	flow: Flow;
 	currentStepRoute: string;
 }
+const debug = debugFactory( 'calypso:landing:stepper:analytics' );
+export const TRACKING_LOCAL_STORAGE_KEY = 'signed_up_tracked';
 
-export const useSignUpStartTracking = ( { flow, currentStepRoute }: Props ) => {
-	const steps = flow.useSteps();
+const setTrackingState = ( id: string ) => {
+	const data = JSON.parse( localStorage.getItem( TRACKING_LOCAL_STORAGE_KEY ) || '{}' );
+	const newState = {
+		...data,
+		[ id ]: true,
```

Comment:
I have a concern here about the way we handle users who revisit a flow multiple times, especially across a long period of time. By setting this flag as `true`, we only know they did this before, but we don't know when.

Can we update the logic to drop a timestamp? That way we can have `wasAlreadyTracked` do a timestamp check to record a new event if it occurs after 12 or 24 hours.

Reply:
I am trying to think of a solution different from expiration logic because I think we can have users migrating multiple sites.